config/routes.rb

```diff
@@ -1,8 +1,9 @@
 Rails.application.routes.draw do
+  get 'posts/:id', to: 'posts#show', as: 'post'
   resources :users, only: [:index, :show, :new] do
-  resources :posts, only:[:index, :show , :new] do
-  resources :comments, only: [:new, :create]
-  resources :likes, only: [:new, :create]
+    resources :posts, only: [:index, :show, :new, :create] do
+      resources :comments, only: [:new, :create]
+        resources :likes, only: [:new, :create]
```

Comment:
[REQUIRED] You should have a likes endpoint for only create. We don't want to waste resources on an endpoint that won't be put to use.

Reply:
im using new route for creating a like so removing create endpoint